src/components/periodic-report/handlers/sync-periodic-report-to-project.handler.ts

```diff
@@ -0,0 +1,115 @@
+import { isNull, node, relation } from 'cypher-query-builder';
+import { DateTime } from 'luxon';
+import { Session } from '../../../common';
+import {
+  DatabaseService,
+  EventsHandler,
+  IEventHandler,
+  ILogger,
+  Logger,
+} from '../../../core';
+import { Project } from '../../project';
+import { ProjectUpdatedEvent } from '../../project/events';
+import { CreatePeriodicReport } from '../dto';
+import { ReportType } from '../dto/type.enum';
+import { PeriodicReportService } from '../periodic-report.service';
+
+type SubscribedEvent = ProjectUpdatedEvent;
+
+@EventsHandler(ProjectUpdatedEvent)
+export class SyncPeriodicReportToProject
+  implements IEventHandler<SubscribedEvent> {
+  constructor(
+    private readonly db: DatabaseService,
+    private readonly periodicReports: PeriodicReportService,
+    @Logger('periodicReport:sync-project') private readonly logger: ILogger
+  ) {}
+
+  async handle(event: SubscribedEvent) {
+    this.logger.debug('Project mutation, syncing periodic reports', {
+      ...event,
+      event: event.constructor.name,
+    });
+
+    const project = event.updated;
+    const previous = event.previous;
+
+    const isDateRangeChanged =
+      project.mouStart !== previous.mouStart ||
+      project.mouEnd !== previous.mouEnd;
```

Comment:
These are date objects and should not be compared with identity. 
https://moment.github.io/luxon/docs/manual/math.html#comparing-datetimes
```tsx
d1.toMillis() === d2.toMillis() // are d1 and d2 the same instant in time?
+d1 === +d2 // same test, using object coercion
```